docs/data-sources/port.md

```diff
@@ -0,0 +1,45 @@
+---
+page_title: "Equinix Metal: precreated_port"
+subcategory: ""
+description: |-
+  Fetch device ports
+---
+
+# metal_port
+
+Use this data source to read ports of existing devices. You can read port by either its UUID, or by a device UUID and port name.
+
+## Example Usage
+
+Create a device and read it's eth0 port to the datasource.
+
+```hcl
+locals {
+  project_id = "<UUID_of_your_project>"
+}
+
+resource "metal_device" "test" {
+  hostname         = "tfacc-test-device-port"
+  plan             = "c3.medium.x86"
+  facilities       = ["sv15"]
+  operating_system = "ubuntu_20_04"
+  billing_cycle    = "hourly"
+  project_id       = local.project_id
+}
+
+data "metal_port" "test" {
+    device_id        = metal_device.test.id
+    port_name        = "eth0"
+}
+```
+
+## Argument Reference
+
+* `id` - (Required) ID of the port to read, conflicts with device_id.
+* `device_id` - (Required) 4 or 6, depending on which block you are looking for.
```

Comment:
I considered suggesting that we call this `hardware` to match the API, but I'm not sure how we would be able to determine if the user means "hardware is a reservation" or "hardware is a device instance" or "hardware is something else".

It makes sense to use "device_id" + "port.name" for lookups, and if we need more lookups in the future we'll have to add more lookup fields.

Should we set `device_id` based on the `hardware.href` returned (assuming there href basename is the id)?  Perhaps we need to query `/ports/{id}?include=hardware` to gather the needed information.

Reply:
sorry about the doc string for the `device_id`.

I'm a bit confused here. The hardware ID is different than the device ID, and the href link in hardware is not GETtable. I get 404 getting e..g /hardware/8eec90d7-62c4-47c2-b541-b0a0a3f1c428.

I can see the hardware resource when I GET "/ports/9ea27e03-8c5f-4a2a-ac6e-466cb1076c36?include=hardware". It has the device in `instance` attribute. However, we don't have a struct for the "hardware" resource here or in packngo.